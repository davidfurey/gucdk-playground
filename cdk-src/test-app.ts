import { AccessScope } from '@guardian/cdk/lib/constants/access';
import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack } from '@guardian/cdk/lib/constructs/core';
import { GuPlayApp } from '@guardian/cdk/lib/patterns/ec2-app';
import type { App } from 'aws-cdk-lib';
import { InstanceClass, InstanceSize, InstanceType } from 'aws-cdk-lib/aws-ec2';

export class TestApp extends GuStack {
	private readonly app: string = 'test-app';

	constructor(scope: App, id: string, props: GuStackProps) {
		super(scope, id, props);
		new GuPlayApp(this, {
			app: this.app,
			access: { scope: AccessScope.PUBLIC },
			instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MEDIUM),
			certificateProps: {
				domainName: 'prod-guardian.com',
			},
			monitoringConfiguration: {
				snsTopicName: 'alerts-topic-for-my-team',
				http5xxAlarm: {
					tolerated5xxPercentage: 1,
				},
				unhealthyInstancesAlarm: true,
			},
			scaling: { minimumInstances: 1 },
			userData: {
				distributable: {
					fileName: 'app-name.deb',
					executionStatement: `dpkg -i /app-name/app-name.deb`,
				},
			},
		});
	}
}
