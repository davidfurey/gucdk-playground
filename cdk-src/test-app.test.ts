import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { TestApp } from './test-app';

describe('The TestApp stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const stack = new TestApp(app, 'TestApp', {
			stack: 'test-stack',
			stage: 'TEST',
		});
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
