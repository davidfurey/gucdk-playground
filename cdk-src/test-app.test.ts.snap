// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TestApp stack matches the snapshot 1`] = `Object {}`;
